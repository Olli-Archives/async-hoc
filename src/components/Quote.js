import React from 'react';
import PropTypes from 'prop-types';

export default function Quote({ character, quote, image }) {

  return (
    <div>
      <h2>{character}</h2>
      <img src={image}/>
      <p>{quote}</p>
    </div>
  );

}

Quote.propTypes = {
  character: PropTypes.string.isRequired,
  quote: PropTypes.string.isRequired,
  image: PropTypes.string.isRequired
}
;
