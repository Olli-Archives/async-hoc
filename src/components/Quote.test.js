import React from 'react';
import { shallow } from 'enzyme';
import Quote from './Quote';

describe('Quote component', ()=>{
  it('renders character image, title, and quote', ()=>{
    const wrapper = shallow(<Quote image='test-image' character='bender' quote='test quote'/>);
    expect(wrapper).toMatchSnapshot();
  });
})
;
