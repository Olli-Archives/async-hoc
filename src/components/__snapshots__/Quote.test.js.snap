// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Quote component renders character image, title, and quote 1`] = `
<div>
  <h2>
    bender
  </h2>
  <img
    src="test-image"
  />
  <p>
    test quote
  </p>
</div>
`;
